doc/changelog.md

```diff
@@ -1,37 +1,131 @@
-# Release 0.3.3-dev
+# Release 0.4.0
 
 <h3>New features</h3>
 
-* A mitigation dialect (MLIR) was added. It initially contains a Zero Noise Extrapolation (ZNE) operation,
-  with a lowering to a global folded circuit.
-  [(#324)](https://github.com/PennyLaneAI/catalyst/pull/324)
+* Catalyst is now accessible directly within the PennyLane user interface,
+  once Catalyst is installed, allowing easy access to Catalyst just-in-time
+  functionality.
+
+  Through the use of the `qml.qjit` decorator, entire workflows can be JIT
+  compiled down to a machine binary on first-function execution, including both quantum
+  and classical processing. Subsequent calls to the compiled function will execute
+  the previously-compiled binary, resulting in significant performance improvements.
+
+  ```python
+  import pennylane as qml
+
+  dev = qml.device("lightning.qubit", wires=2)
+
+  @qml.qjit
+  @qml.qnode(dev)
+  def circuit(theta):
+      qml.Hadamard(wires=0)
+      qml.RX(theta, wires=1)
+      qml.CNOT(wires=[0, 1])
+      return qml.expval(qml.PauliZ(wires=1))
+  ```
+
+  ```pycon
+  >>> circuit(0.5)  # the first call, compilation occurs here
+  array(0.)
+  >>> circuit(0.5)  # the precompiled quantum function is called
+  array(0.)
+  ```
+
+  Currently, PennyLane supports the [Catalyst hybrid compiler](https://github.com/pennylaneai/catalyst)
+  with the `qml.qjit` decorator, which directly aliases Catalyst's `catalyst.qjit`.
+
+  In addition to the above `qml.qjit` integration, the following native PennyLane functions can now
+  be used with the `qjit` decorator: `qml.adjoint`, `qml.ctrl`, `qml.grad`, `qml.jacobian`,
+  `qml.vjp`, `qml.jvp`, and `qml.adjoint`, `qml.while_loop`, `qml.for_loop`, `qml.cond`. These will
+  alias to the corresponding Catalyst functions when used within a `qjit` context.
+
+  For more details on these functions, please refer to the
+  [PennyLane compiler documentation](https://docs.pennylane.ai/en/latest/introduction/compiling_workflows.html).
```

Comment:
@josh146 The compiler module doc page is also a good target to be linked here. 
Perhaps, we can update this line as follows:
 ```suggestion
  [PennyLane compiler documentation](https://docs.pennylane.ai/en/stable/introduction/compiling_workflows.html) and
  [`qml.compiler`](https://docs.pennylane.ai/en/stable/code/qml_compiler.html).
```

Would you also like to use "stable" instead of "latest" for the docs?